lib/jsdom/living/nodes/HTMLSelectElement-impl.js

```diff
@@ -171,14 +171,20 @@ class HTMLSelectElementImpl extends HTMLElementImpl {
   }
 
   set value(val) {
+    let updated = false;
     for (const option of this.options) {
+      updated = true;
       if (option.value === val) {
         option._selectedness = true;
         option._dirtyness = true;
       } else {
         option._selectedness = false;
       }
```

Comment:
It seems better to call `option._modified()` here, rather than mucking with `_version` ourselves. This is particularly important as it updates `option._version` as well. See the `selected` setter of `HTMLOptionElementImpl`. (The `askForAReset` bit doesn't seem necessary however.)

Reply:
`node._modified()` inherits `O(n)` complexity from `node._clearMemoizedQueries()`, where `n` is the number of ancestor nodes.

This results in calling `option._modified()` having `O(n*m)` complexity, where `n` is the number of option elements and `m` is the number of ancestor nodes from the `option` element.

Reply:
We can assume that `n` is small enough so that we don't have to worry about it. I don't believe the complexity tradeoff is worth it here.